cli/ballerina-cli/src/main/java/io/ballerina/cli/task/RunNativeImageTestTask.java

```diff
@@ -345,4 +402,229 @@ private String getClassPath(JBallerinaBackend jBallerinaBackend, Package current
         return classPath.toString();
     }
 
+    private boolean modifyJarForFunctionMock(TestSuite testSuite, Target target, String moduleName,
+                                             Map<String, String> functionMockModuleMapping) throws IOException {
+        String mainJarName = testSuite.getOrgName() + HYPHEN + moduleName + HYPHEN +
+                testSuite.getVersion() + JAR_EXTENSION;
+        String testJardName = testSuite.getOrgName() + HYPHEN + moduleName + HYPHEN +
+                testSuite.getVersion() + HYPHEN + TESTABLE + JAR_EXTENSION;
+        String modJar = testSuite.getOrgName() + HYPHEN + moduleName + HYPHEN + testSuite.getVersion() + HYPHEN +
+                MODIFIED + JAR_EXTENSION;
+        if (testSuite.getMockFunctionNamesMap().isEmpty()) {
+            functionMockModuleMapping.put(mainJarName, null);
```

Comment:
Let's return an empty map here instead of adding null as the value. We can use the `Map.hasKey` to check if there are any functions to mock.